website/blog/2024/06-24-proxy-management-in-crawlee/index.md

```diff
@@ -0,0 +1,104 @@
+slug: proxy-management-in-crawlee
+title: 'How Crawlee uses tiered proxies to avoid getting blocked'
+tags: [proxy]
+description: 'Find out how Crawlee’s tiered proxy system rotates between different types of proxies to control web scraping costs and avoid getting blocked.'
+image: ./img/tiered-proxies.png
+author: Saurav Jain
+authorTitle: Developer Community Manager @ Crawlee
+authorURL: https://github.com/souravjain540
+authorImageURL: https://avatars.githubusercontent.com/u/53312820?v=4
+authorTwitter: sauain
+---
+
+Hello Crawlee community,
+
+We are back with another blog, this time explaining how Crawlee rotates proxies and prevents crawlers from getting blocked.
+
+Proxies vary in quality, speed, reliability, and cost. There are a [few types of proxies](https://blog.apify.com/types-of-proxies/), such as datacenter and residential proxies. Datacenter proxies are cheaper but, on the other hand, more prone to getting blocked, and vice versa with residential proxies.
+
+It is hard for developers to decide which proxy to use while scraping data. We might get blocked if we use [datacenter proxies](https://blog.apify.com/datacenter-proxies-when-to-use-them-and-how-to-make-the-most-of-them/) for low-cost scraping, but residential proxies are sometimes too expensive for bigger projects. Developers need a system that can manage both costs and avoid getting blocked. To manage this, we recently introduced `TieredProxies` in Crawlee. Let’s take a look at it.
+
+:::note
+
+If you like reading this blog, we would be really happy if you gave [Crawlee a star on GitHub!](https://github.com/apify/crawlee/)
+
+:::
+
+## What are tiered proxies?
+
+Tiered proxies are a method of organizing and using different types of proxies based on their quality, speed, reliability, and cost. Tiered proxies allow you to rotate between a mix of proxy types to optimize your scraping activities.
+
+**Define proxy tiers**: You categorize your proxies into different tiers based on their quality. For example:
+
+- **High-tier proxies**: Fast, reliable, and expensive. Best for critical tasks where you need high performance.
+- **Mid-tier proxies**: Moderate speed and reliability. A good balance between cost and performance.
+- **Low-tier proxies**: Slow and less reliable but cheap. Useful for less critical tasks or high-volume scraping.
+
+## Features:
+
+- **Tracking errors**: The system monitors errors (e.g. failed requests, retries) for each domain.
+- **Adjusting tiers**: Higher-tier proxies are used if a domain shows more errors. Conversely, if a domain performs well with a high-tier proxy, the system will occasionally test lower-tier proxies. If successful, it continues using the lower tier, optimizing costs.
+- **Forgetting old errors**: Old errors are given less weight over time, allowing the system to adjust tiers dynamically as proxies' performance changes.
+
+### Working
```

Comment:
I hope it is 

jokes aside, the paragraphs from `Working` to `Implementation` seem very bare-bones. The feature is IMO very simple to explain, I don't think we need to inflate the blog post. It can be short, there is just not enough content to talk about.

Reply:
the need to add these things are mostly because there is already a lot of content on Scrapy rotating proxies boasting their features and we don't want users to feel our feature does not do those things, when it actually does.

I understand its not fully developer oriented part, but in the end it is a blog not exactly docs, so we can and I think should be a little more explanation here :)

Reply:
I understand the need for good SEO content, but I'm not sure this is it. E.g.:

**Structure** - the structure of what? and why is the **Structure** heading nested under the **Features**? Also "Tiered Array" doesn't tell me anything (plus the description is wrong, the type is `string[][]`, i.e. each element in the array is an _array of strings_ - proxy URLs).

In general, I'm all for the "show don't tell" approach - the example shows much more to me than any amount of text.

Reply:
I'm still hating on this part 😄 But as long as you fix the wrong type in the `Structure` paragraph, it's a go from me - I don't want to block this for too long.

Reply:
I got your point; thanks for putting in the effort. 🫶

Reply:
i removed the structure and tiered array, both were not giving any value. 👍 

also corrected the headings!

Reply:
@barjin if its okay let's release it today? :)

Reply:
Sure, it's a go from me 👍🏽